modules/service/src/main/scala/lucuma/odb/service/SequenceService.scala

```diff
@@ -434,46 +400,21 @@ object SequenceService {
           $step_type
       """.command
 
-    val SelectCompletedStepRecordsForObs: Query[Observation.Id, (Atom.Id, NonNegShort, Step.Id)] =
+    val SelectCompletedStepRecordsForObs: Query[Observation.Id, (Atom.Id, NonNegShort, SequenceType, Step.Id)] =
       (sql"""
         SELECT
           a.c_atom_id,
           a.c_step_count,
+          a.c_sequence_type,
           s.c_step_id
         FROM t_step_record s
         INNER JOIN t_atom_record a ON a.c_atom_id = s.c_atom_id
         WHERE """ ~> sql"""a.c_observation_id = $observation_id AND s.c_completed IS NOT NULL ORDER BY s.c_completed"""
-      ).query(atom_id *: int2_nonneg *: step_id)
+      ).query(atom_id *: int2_nonneg *: sequence_type *: step_id)
 
     def encodeColumns(prefix: Option[String], columns: List[String]): String =
       columns.map(c => s"${prefix.foldMap(_ + ".")}$c").intercalate(",\n")
 
-// TODO: I would like to write the insert step config logic once, parameterized
-// by the table name, columns, and the encoder.  I think I'm being thwarted by
-// scala.quoted / macro issues?
-//
```

Comment:
Do you still need a workaround for this?

Reply:
I decided I could live with the way I left it.